feat(improvements): add base improvements to the project

grski opened the pull request:
… docs, formatting, example Makefile, CI/CD, pre-commit
Hey guys, as per our talk.

While I do want the feature to get shipped, I have various concerns regarding the general quality of the code. Before proceeding to the implementation, I deemed it necessary to introduce some improvements in various areas, which include:
1. env management
2. docs
3. formatting/styling
4. ci/cd
5. dependabot
so on, so forth.

This is just for starters.

I'll get @hkorzeniewski working on this more.

grski commented:
Before this one gets merged, you should take a look at: https://github.com/th33ngineers/reliableGPT/pull/1

ishaan-jaff commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,38 @@
+      pull-requests: write
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
```
What tests does this run ? Is there a way to make it run python tests ?

grski replied:
it check outs the repo, then using `make` utility runs the tests

ishaan-jaff commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,12 @@
+repos:
```
what does this do ?

grski replied:
this is for https://pre-commit.com/
pip install pre-commit on your local machine 
this package runs certain steps on your codebase before you commit files 
so once you made changes, it automatically, in this case, runs formatting/linting on the changes you've made

ishaan-jaff commented on Makefile:
```diff
@@ -0,0 +1,18 @@
+PATH  := $(PATH)
```
can I get a summary on what this does roughly ? Is it running through lint + tests

grski replied:
make is a na utility that executes stuff you want 
so instead of writing "poetry run ruff check . --fix" for linting each time you just write "make ruff"

grski reviewed (commented):
answered

grski commented:
@adamjanas good first open source contribution :)

grski commented:
@ishaan-jaff do you think it's fine to approve & merge these changes? If yes we'd like to do so and move further

grski commented:
Okay, just added a couple of things with @hkorzeniewski 
These changes above introduce:
1. consistent naming of the files
2. moved tests to one directory outside the code
3. removed files that should not be committed like builds (next step add automatic build & deployment to ci/cd)
4. CI/CD through github actions with automated linting and tests (some tests need fixing, fixed couple of them already). Linting/formatting also needs some love, we fixed what we could
5. Makefile utility
6. Dependabot config (updates the requirements automatically)
7. fixed for the most part formatting issues
8. added packages that were missing in poetry config but were needed to get this to work

It's a big one, I'm keen to hear your opinion.
